app/scripts/lib/util.js

```diff
@@ -144,6 +144,31 @@ function removeListeners (listeners, emitter) {
   })
 }
 
+function fetchWithTimeout ({ timeout = 120000 } = {}) {
+  return function _fetch (url, opts) {
+    return async (resolve, reject) => {
```

Comment:
I think you may have made a mistake here - there's a function wrapper left behind with `resolve` and `reject` that aren't used.

The `_fetch` function is what should be async, minus this second function wrapper.

Reply:
Mistake I did make